spring-cloud-common-security-config-web/src/main/java/org/springframework/cloud/common/security/OAuthSecurityConfiguration.java

```diff
@@ -106,272 +108,440 @@
-		return providerManager;
-	}
-
-	@Bean
-	protected OAuth2TokenUtilsService oauth2TokenUtilsService() {
-		return new DefaultOAuth2TokenUtilsService(this.oauth2AuthorizedClientService);
-	}
-
-	@EventListener
-	public void handleOAuth2AuthenticationFailureEvent(
-			AbstractAuthenticationFailureEvent authenticationFailureEvent) {
-		logger.warn("An authentication failure event occurred while accessing a REST resource that requires authentication.",
-				authenticationFailureEvent.getException());
-	}
-
-	protected String dashboard(String path) {
-		return this.authorizationProperties.getDashboardUrl() + path;
-	}
-
-	protected static class BrowserDetectingContentNegotiationStrategy extends HeaderContentNegotiationStrategy {
-
-		@Override
-		public List<MediaType> resolveMediaTypes(NativeWebRequest request)
-				throws HttpMediaTypeNotAcceptableException {
-			final List<MediaType> supportedMediaTypes = super.resolveMediaTypes(request);
-
-			final String userAgent = request.getHeader(HttpHeaders.USER_AGENT);
-			if (userAgent != null && userAgent.contains("Mozilla/5.0")
-					&& !supportedMediaTypes.contains(MediaType.APPLICATION_JSON)) {
-
-				return Collections.singletonList(MediaType.TEXT_HTML);
-			}
-			return Collections.singletonList(MediaType.APPLICATION_JSON);
-		}
-	}
-
-	@Bean
-	OAuth2AccessTokenResponseClient<OAuth2PasswordGrantRequest> oAuth2PasswordTokenResponseClient() {
-		return new DefaultPasswordTokenResponseClient();
-	}
-
-	private String calculateDefaultProviderId() {
-		if (this.authorizationProperties.getDefaultProviderId() != null) {
-			return this.authorizationProperties.getDefaultProviderId();
-		}
-		else if (this.oauth2ClientProperties.getRegistration().size() == 1) {
-			return this.oauth2ClientProperties.getRegistration().entrySet().iterator().next().getKey();
-		}
-		else if (this.oauth2ClientProperties.getRegistration().size() > 1
-				&& !StringUtils.hasText(this.authorizationProperties.getDefaultProviderId())) {
-			throw new IllegalStateException("defaultProviderId must be set if more than 1 Registration is provided.");
-		}
-		else {
-			throw new IllegalStateException("Unable to retrieve default provider id.");
-		}
-	}
+    private static final org.slf4j.Logger logger = LoggerFactory.getLogger(OAuthSecurityConfiguration.class);
 
+    @Autowired
+    protected OAuth2ClientProperties oauth2ClientProperties;
+
+    @Autowired
+    protected SecurityStateBean securityStateBean;
+
+    @Autowired
+    protected SecurityProperties securityProperties;
+
+    @Autowired
+    protected ApplicationEventPublisher applicationEventPublisher;
+
+    @Autowired
+    protected AuthorizationProperties authorizationProperties;
+
+    @Autowired
+    protected OAuth2ResourceServerProperties oAuth2ResourceServerProperties;
+
+    @Autowired
+    @Qualifier("plainOauth2UserService")
+    protected OAuth2UserService<OAuth2UserRequest, OAuth2User> plainOauth2UserService;
+
+    @Autowired
+    @Qualifier("oidcUserService")
```

Comment:
The use of `@Bean(name = "xyz") + @Qualifier("xyz") + field named "xyz"` is redundant as the bean name is the qualifier. Is there a reason for adding this other than an attempt to prevent breaking if someone renames the bean? That strategy usually does not help and tests should alert us if it breaks. I would prefer to remove the redundant specifiers. Wdyt?  (this applies to all locations in this auto-config)

Reply:
Removed